nemo/collections/tts/models/speechllm/megatron_t5_speechllm_model.py

```diff
@@ -801,6 +833,40 @@ def on_validation_epoch_start(self) -> None:
     def training_step(self, dataloader_iter, batch_idx):
         self._optimizer.zero_grad()
         batch = next(dataloader_iter)
+
+        # apply classifier-free guidance by replacing input question tokens with [UNK].
+        if self._rng.random() < self.question_free_guidance_prob:
```

Comment:
Can we skip `self._rng.random()` if tranining with cfg is disabled?

Reply:
yeah, better to skip `self._rng.random()` at each training step when cfg is disabled to save some computations. Let me add something like below,
```suggestion
        if self.question_free_guidance_prob > 0.0:
            if self._rng.random() < self.question_free_guidance_prob:
                ...
            else:
                # recover to original alignment loss config.
                self.frozen_model.enc_dec_model.use_alignment_loss = self.use_alignment_loss
```